Sources/Sentry/SentryTransportFactory.m

```diff
@@ -29,9 +29,18 @@ @implementation SentryTransportFactory
 {
     NSURLSessionConfiguration *configuration =
         [NSURLSessionConfiguration ephemeralSessionConfiguration];
-    NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration
-                                                          delegate:options.urlSessionDelegate
-                                                     delegateQueue:nil];
+    
+    NSURLSession *session;
+    
+    if (options.urlSession) {
+     
+        session = options.urlSession;
+    } else {
+        session = [NSURLSession sessionWithConfiguration:configuration
+                                                              delegate:options.urlSessionDelegate
+                                                        delegateQueue:nil];
```

Comment:
m: You could move lines 30, 31 in here.

Reply:
Moved `ephemeralSessionConfiguration` inside else statement.